Assets/Scripts/SlownessOnImpact.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SlownessOnImpact : MonoBehaviour {
+
+	public float slownessFactor = 10;
+	void Start () {
+		
+	}
+	
+	// Update is called once per frame
+	void Update () {
+	}
+
+	private void OnTriggerEnter2D(Collider2D e)
+	{
+		if(e.gameObject.GetComponent<PathMovement> ()){
+			e.gameObject.GetComponent<PathMovement> ().speed = e.gameObject.GetComponent<PathMovement>().speed / slownessFactor;
+		}
+	}
+
+	private void OnTriggerExit2D(Collider2D e)
+	{
+		if(e.gameObject.GetComponent<PathMovement> ()){
+			e.gameObject.GetComponent<PathMovement> ().speed = e.gameObject.GetComponent<PathMovement>().speed * slownessFactor;
```

Comment:
Hmm, good point. But I'm planing to make it so that even other game objects slowdown whilst visiting a blackhole (like they do). I'm not sure how your speed reverter works. Can you change this script to work with it?